axlearn/common/aot_compilation.py

```diff
@@ -0,0 +1,90 @@
+"""Utilities for ahead-of-time compilation.
+
+Reference:
+https://docs.google.com/document/d/1Y5IdmvAZA7UtMHAWkRh8k2PscVoG5FvMH9-E6hygsyY/
+"""
+import os
+
+os.environ["JAX_PLATFORMS"] = "cpu"
+os.environ["TPU_SKIP_MDS_QUERY"] = "1"
+
+import copy
+from dataclasses import dataclass
+from typing import Callable, Dict
+
+import jax
+jax.config.update("jax_platforms", "cpu")
```

Comment:
Since this PR sets the topology explicitly when we instantiate SpmdTrainer , could you comment on why setting the platform to cpu is needed?